src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -234,11 +266,6 @@ protected AnalyzerDriver(ImmutableArray<DiagnosticAnalyzer> analyzers, AnalyzerM
 
             this.CompilationEventQueue = new AsyncQueue<CompilationEvent>();
             this.DiagnosticQueue = new AsyncQueue<Diagnostic>();
-            _queueRegistration = cancellationToken.Register(() =>
```

Comment:
This is removed because completing the queues asynchronously can cause a failure if a non-quite-canceled action attempts to enqueue a compilation event or a diagnostic.